NextGenConnect/JSON to HL7v2.xml

```diff
@@ -83,35 +83,257 @@ for (var i = 0; i &lt; getArrayOrXmlLength(patient[&apos;identifier&apos;]); i++
 }
 tmp[&apos;PID&apos;][&apos;PID.3&apos;] = ids.join(&apos;~&apos;);
 
-
-/*
 //PID-5 patient name
-tmp[&apos;PID&apos;][&apos;PID.5&apos;][&apos;PID.5.1&apos;] = patient[&apos;name&apos;][0][&apos;family&apos;][&apos;@value&apos;].toString();
-tmp[&apos;PID&apos;][&apos;PID.5&apos;][&apos;PID.5.2&apos;] = patient[&apos;name&apos;][0][&apos;given&apos;][&apos;@value&apos;].toString();
-tmp[&apos;PID&apos;][&apos;PID.5&apos;][&apos;PID.5.3&apos;] = &quot;&quot;; //not in XML
+tmp[&apos;PID&apos;][&apos;PID.5&apos;][&apos;PID.5.1&apos;] = patient[&apos;name&apos;][0][&apos;family&apos;];
+tmp[&apos;PID&apos;][&apos;PID.5&apos;][&apos;PID.5.2&apos;] = patient[&apos;name&apos;][0][&apos;given&apos;][0]; //Why is this stored in an array in sample JSON message?
+tmp[&apos;PID&apos;][&apos;PID.5&apos;][&apos;PID.5.7&apos;] = &quot;L&quot;;  // Legal Name		
+
+//PID-6 Mothers Maiden Name -- ??? Try to grab value ?
+
+//PID-7 Birthdate 
+
+var birthDate = patient[&apos;birthDate&apos;];
+
+birthDate.replace(&apos;-&apos;,&apos;&apos;); // Replaces -&apos;s in birthDate
+
+tmp[&apos;PID&apos;][&apos;PID.7&apos;][&apos;PID.7.1&apos;] = birthDate;
+
+
+//PID-8 Administrative Sex -- Needs to be edited for actualy values F/M
+
+if(patient[&apos;gender&apos;] == &apos;female&apos;){  
+	tmp[&apos;PID&apos;][&apos;PID.8&apos;][&apos;PID.8.1&apos;] = &apos;F&apos;;
+}
+else if(patient[&apos;gender&apos;] == &quot;male&quot;){
+	tmp[&apos;PID&apos;][&apos;PID.8&apos;][&apos;PID.8.1&apos;] = &apos;M&apos;;
+	}
+
+else{
+	tmp[&apos;PID&apos;][&apos;PID.8&apos;][&apos;PID.8.1&apos;] = &apos;U&apos;;
+}
+
+//PID-10 Race  -- ??? Try to grab value ?
+
 
-tmp[&apos;PID&apos;][&apos;PID.7&apos;][&apos;PID.7.1&apos;] = patient[&apos;birthDate&apos;][&apos;@value&apos;].toString();
-tmp[&apos;PID&apos;][&apos;PID.8&apos;][&apos;PID.8.1&apos;] = patient[&apos;gender&apos;][&apos;@value&apos;].toString();
 
 //PID-11 address
-tmp[&apos;PID&apos;][&apos;PID.11&apos;][&apos;PID.11.1&apos;] = patient[&apos;address&apos;][&apos;line&apos;][&apos;@value&apos;].toString();
-tmp[&apos;PID&apos;][&apos;PID.11&apos;][&apos;PID.11.7&apos;] = patient[&apos;address&apos;][&apos;use&apos;][&apos;@value&apos;].toString();
-tmp[&apos;PID&apos;][&apos;PID.11&apos;][&apos;PID.11.3&apos;] = patient[&apos;address&apos;][&apos;city&apos;][&apos;@value&apos;].toString();
-tmp[&apos;PID&apos;][&apos;PID.11&apos;][&apos;PID.11.4&apos;] = patient[&apos;address&apos;][&apos;state&apos;][&apos;@value&apos;].toString();
-tmp[&apos;PID&apos;][&apos;PID.11&apos;][&apos;PID.11.5&apos;] = patient[&apos;address&apos;][&apos;postalCode&apos;][&apos;@value&apos;].toString();
+tmp[&apos;PID&apos;][&apos;PID.11&apos;][&apos;PID.11.1&apos;] = patient[&apos;address&apos;][0][&apos;line&apos;][0];
```

Comment:
Taking only the first line might leave an incomplete address, like if the patient lives in an apartment or something. Can you do a .join with a comma like in PID-3 so we get all the lines?

Reply:
I don't really see the point of joining these. Taking the first line would signify the primary address, would it not? The second line being a secondary... etc. I'd assume a patient with an apartment address would be the same.

Reply:
At least as far as I have seen in every sample fhir patient resource I've looked at.

Reply:
I think you're confusing patient['address'], which is an array of address objects with patient['address'][n]['line'], which is an array of strings. Each of the address objects in patient['address'] can have an array of lines that looks like this:

Line 1: 1234 Main Street
Line 2: Apartment 502

If we just take the first address line, which is what you're doing if you only get patient['address'][0]['line'][0], you won't get the apartment number, suite, or whatever else people put on the subsequent address lines.

Reply:
I didn't do this yet however I wanted to show you this:
"address": [ {
                "extension": [ {
                    "url": "http://hl7.org/fhir/StructureDefinition/geolocation",
                    "extension": [ {
                        "url": "latitude",
                        "valueDecimal": 42.08145721749866
                    }, {
                        "url": "longitude",
                        "valueDecimal": -87.69894004417375
                    } ]
                } ],
                "line": [ "658 Lueilwitz Spur Apt 55" ],
                "city": "Chicago",
                "state": "IL",
                "postalCode": "60007",
                "country": "US"
            } ]
Does this change your thinking at all?

Reply:
It really depends on whether we store it in different elements of the array, which wouldn't make any sense to me. Pretty much all the fhir examples list the "line" out like this.

Reply:
This is what it would look like if the patient had multiple addresses, each with multiple address lines (the [] signifies an array in JSON):
"address": [
    {
      "use": "home",
      "line": [
        "Van Egmondkade 23",
        "Apartment 23"
      ],
      "city": "Amsterdam",
      "postalCode": "1024 RJ",
      "country": "NLD"
    },
    {
      "use": "work",
      "line": [
        "Van Egmondkade 24",
        "Suite 101"
      ],
      "city": "Amsterdam",
      "postalCode": "1024 RJ",
      "country": "NLD"
    }
  ]